src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -395,6 +395,13 @@ private async Task ProduceEndAwaited()
         await _bodyOutput.FlushAsync(default);
     }
 
+    // Response trailers, reset, and GOAWAY are only on HTTP/2+ and require IIS support
+    // that is only available on Win 11/Server 2022 or later.
+    private static readonly bool isCompatibleOsVersion = OperatingSystem.IsWindowsVersionAtLeast(10, 0, 20348, 0);
+    protected bool AdvancedHttp2FeaturesSupported => isCompatibleOsVersion &&
```

Comment:
nit: Change AdvancedHttp2FeaturesSupported from a property to a method

It's a small change to indicate there is some cost in calling it (I'm guessing the native method call has some small overhead)